aggregator/src/aggregator/aggregation_job_creator.rs

```diff
@@ -2965,23 +3400,36 @@ mod tests {
         .unwrap();
 
         // Verify that all reports we saw a report aggregation for are scrubbed.
-        let all_seen_report_ids: HashSet<_> = agg_jobs_and_report_ids
+        let all_seen_report_ids: HashSet<(A::AggregationParam, _)> = agg_jobs_and_report_ids
             .iter()
-            .flat_map(|(_, report_aggs)| report_aggs.iter().map(|ra| ra.report_id()))
+            .flat_map(|(agg_job, report_aggs)| {
+                report_aggs.iter().map(|ra| {
+                    (
+                        (agg_job.aggregation_parameter() as &A::AggregationParam).clone(),
+                        ra.report_id(),
+                    )
+                })
+            })
             .collect();
-        for report_id in &all_seen_report_ids {
-            tx.verify_client_report_scrubbed(task_id, report_id).await;
+        for (agg_param, report_id) in &all_seen_report_ids {
+            if agg_param.type_id() == ().type_id() {
+                tx.verify_client_report_scrubbed(task_id, report_id).await;
+            }
         }
 
-        // Verify that all reports we did not see a report aggregation for are not scrubbed. (We do
-        // so by reading the report, since reading a report will fail if the report is scrubbed.)
-        for report_id in want_ra_states.keys() {
-            if all_seen_report_ids.contains(report_id) {
-                continue;
+        // Verify that if the aggregation parameter is (), all reports we saw a report aggregation
```

Comment:
I think this comment goes above the block above which checks if the report is scrubbed.